optimum/intel/neural_compressor/utils.py

```diff
@@ -109,3 +114,240 @@ def load_quantized_model(checkpoint_dir_or_file: str, model: torch.nn.Module, **
+        self.traced_model = traced_model
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
+        import copy
+        import os
+        from pathlib import Path
+
+        from transformers.configuration_utils import PretrainedConfig
+        from transformers.dynamic_module_utils import get_class_from_dynamic_module
+        from transformers.utils import TRANSFORMERS_CACHE, is_offline_mode
+
+        from huggingface_hub import hf_hub_download
+
+        download_kwarg_default = [
+            ("cache_dir", None),
+            ("force_download", False),
+            ("resume_download", False),
+            ("revision", None),
+        ]
+        download_kwargs = {name: kwargs.get(name, default_value) for (name, default_value) in download_kwarg_default}
+
+        config = kwargs.pop("config", None)
+        trust_remote_code = kwargs.pop("trust_remote_code", False)
+        kwargs["_from_auto"] = True
+        hub_kwargs_names = [
+            "cache_dir",
+            "force_download",
+            "local_files_only",
+            "proxies",
+            "resume_download",
+            "revision",
+            "subfolder",
+            "use_auth_token",
+        ]
+        hub_kwargs = {name: kwargs.pop(name) for name in hub_kwargs_names if name in kwargs}
+        if not isinstance(config, PretrainedConfig):
+            kwargs_copy = copy.deepcopy(kwargs)
+            # ensure not to pollute the config object with torch_dtype="auto" - since it's
+            # meaningless in the context of the config object - torch.dtype values are acceptable
+            if kwargs_copy.get("torch_dtype", None) == "auto":
+                _ = kwargs_copy.pop("torch_dtype")
+
+            config, kwargs = AutoConfig.from_pretrained(
+                pretrained_model_name_or_path,
+                return_unused_kwargs=True,
+                trust_remote_code=trust_remote_code,
+                **hub_kwargs,
+                **kwargs_copy,
+            )
+        is_torchscript = getattr(config, "torchscript", None) == "true"
+        if hasattr(config, "auto_map") and cls.__name__ in config.auto_map:
+            if not trust_remote_code:
+                raise ValueError(
+                    f"Loading {pretrained_model_name_or_path} requires you to execute the modeling file in that repo "
+                    "on your local machine. Make sure you have read the code there to avoid malicious use, then set "
+                    "the option `trust_remote_code=True` to remove this error."
+                )
+            if hub_kwargs.get("revision", None) is None:
+                logger.warning(
+                    "Explicitly passing a `revision` is encouraged when loading a model with custom code to ensure "
+                    "no malicious code has been contributed in a newer revision."
+                )
+            class_ref = config.auto_map[cls.__name__]
```

Comment:
why can't we load the model with classes such as `AutoModelForCausalLM` ?

Reply:
We need rewrite __call__ and save_pretrained function, so we only get class here.